test/mongoose_push_web/controllers/api_v1_notification_controller_test.exs

```diff
@@ -17,67 +17,56 @@ defmodule MongoosePushWeb.APIv1NotificationControllerTest do
     assert json_response(conn, 200) == nil
   end
 
-  test "Request.SendNotification.FlatNotification schema without required service field", %{
-    conn: conn
-  } do
-    conn =
-      post(
-        conn,
-        "/v1/notification/666",
-        Jason.encode!(Map.drop(ControllersHelper.flat_request(), ["service"]))
-      )
-
-    assert json_response(conn, 422) == ControllersHelper.missing_field_response("service")
-  end
-
-  test "Request.SendNotification.FlatNotification schema without required body field", %{
-    conn: conn
-  } do
-    conn =
-      post(
-        conn,
-        "/v1/notification/666",
-        Jason.encode!(Map.drop(ControllersHelper.flat_request(), ["body"]))
-      )
-
-    assert json_response(conn, 422) == ControllersHelper.missing_field_response("body")
+  @dropped_fields ["service", "body"]
+  for dropped <- @dropped_fields do
+    test "Request.SendNotification.FlatNotification schema without required #{dropped} field", %{
+      conn: conn
+    } do
+      body =
+        ControllersHelper.flat_request()
+        |> Map.drop([unquote(dropped)])
+        |> Jason.encode!()
```

Comment:
Niiiice, less duplication 😄 

Just one question, maybe I can learn something about elixir here. Why's that `unquote` in `unquote(dropped)` necessary?

Reply:
It's macros magic 😁 We're reading the values from so-called `module attribute` declared with `@`, and because we're kinda injecting it in test body, it needs to be evaluated (back to the string like I wrote the code) and inserted in the place of `unquote(dropped)`. To understand more what this macro stuff is about, you can head to [here](https://elixir-lang.org/getting-started/meta/quote-and-unquote.html), [here](https://www.theerlangelist.com/article/macros_1), or [this answer](https://elixirforum.com/t/i-dont-understand-quote-unquote-why-do-we-need-them/27106/4) on elixir forum almost exactly covering this specific case.